com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -1319,6 +1585,36 @@ static void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier[] colorAtt
             CoreUtils.SetRenderTarget(cmd, colorAttachments, depthAttachment, clearFlag, clearColor);
         }
 
+        static void SetRenderTarget(CommandBuffer cmd, RenderTargetIdentifier[] colorAttachments, RTHandle depthAttachment, ClearFlag clearFlag, Color clearColor)
+        {
+            m_ActiveColorAttachments = colorAttachments;
+            m_ActiveDepthAttachment = depthAttachment.nameID;
+
+            CoreUtils.SetRenderTarget(cmd, m_ActiveColorAttachments, depthAttachment, clearFlag, clearColor);
+            CoreUtils.SetViewport(cmd, depthAttachment);
+        }
+
+        static void SetRenderTarget(CommandBuffer cmd, RTHandle[] colorAttachments, RTHandle depthAttachment, ClearFlag clearFlag, Color clearColor)
+        {
+            if (m_ActiveColorAttachments.Length != colorAttachments.Length)
+            {
+                m_ActiveColorAttachments = new RenderTargetIdentifier[colorAttachments.Length];
+            }
+
+            for (int i = 0; i < colorAttachments.Length; ++i)
+            {
+                m_ActiveColorAttachments[i] = colorAttachments[i].nameID;
+            }
+            m_ActiveDepthAttachment = depthAttachment.nameID;
+
+            CoreUtils.SetRenderTarget(cmd, m_ActiveColorAttachments, depthAttachment, clearFlag, clearColor);
+            foreach (var attachment in colorAttachments)
+            {
+                CoreUtils.SetViewport(cmd, attachment);
+            }
+            CoreUtils.SetViewport(cmd, depthAttachment);
+        }
```

Comment:
That setting of viewport shouldn't be there. Good catch.
A note about this particular call to `SetRenderTarget`: Only `depthAttachment` is an `RTHandle`. `m_ActiveColorAttachments` are `RenderTargetIdentifier` types.
That is not ideal but works because the viewport data in `depthAttachment` would be enough.

I will see if I can call only 
```
CoreUtils.SetRenderTarget(cmd, colorAttachments, depthAttachment, clearFlag, clearColor);
```

Reply:
The issue with calling `public static void SetRenderTarget(CommandBuffer cmd, RTHandle[] colorBuffers, RTHandle depthBuffer, ...)` is that it would require converting `RTHandle[] colorBuffers` to `RenderTargetUtils[]` in `CoreUtils` and I don't think I can do this without allocating memory.

I should still be able to remove the calls to `SetViewport` due to the depth attachment.